### ISSUE TITLE
Convert spark dataframe to h2oframe fails in rsparkling

### ISSUE THREAD

**Issue by joscani** [h2oai/sparkling-water#2377]
- Sparkling Water/PySparkling/RSparkling version 3.32.0.1.2.2.4
- Hadoop Version & Distribution: EMR in aws version 5.23
- Execution mode `YARN-client`, `YARN-cluster`, standalone, local ..

Please also provide us with the full and minimal reproducible code.

Convert a big file to h2oframe fails. 
I start h2ocontext over sparkcontext and all looks good, flow open . 

But if I try 

```r
train_tbl <- spark %>% 
    tbl("rosetta.train_for_models") %>% 
    compute()


train <- hc$asH2OFrame(train_tbl, "train_hex")
```
It fails, But If I use version **3.30.1.2-1-2.4** it works perfectly or only converts a few rows of train_tbl (using version 3.32)  it works.  Any idea?. I tried severaal times and always get the error. 
The log is 

```bash
Error: ai.h2o.sparkling.backend.exceptions.RestApiCommunicationException: H2O node 10.159.129.83:54321 responded with
Status code: 500 : Server Error
Server error: {"__meta":{"schema_version":3,"schema_name":"H2OErrorV3","schema_type":"H2OError"},"timestamp":1603957495646,"error_url":"/3/FinalizeFrame","msg":"\n\nERROR MESSAGE:\n\nDistributedException from ip-10-159-129-83.eu-west-1.compute.internal/10.159.129.83:54321: '1'\n\n","dev_msg":"\n\nERROR MESSAGE:\n\nDistributedException from ip-10-159-129-83.eu-west-1.compute.internal/10.159.129.83:54321: '1'\n\n","http_status":500,"values":{},"exception_type":"water.util.DistributedException","exception_msg":"\n\nERROR MESSAGE:\n\nDistributedException from ip-10-159-129-83.eu-west-1.compute.internal/10.159.129.83:54321: '1'\n\n","stacktrace":["DistributedException from ip-10-159-129-83.eu-west-1.compute.internal/10.159.129.83:54321: '1', caused by java.lang.ArrayIndexOutOfBoundsException: 1","    water.MRTask.getResult(MRTask.java:494)","    water.MRTask.getResult(MRTask.java:502)","    water.MRTask.doAll(MRTask.java:397)","    water.MRTask.doAll(MRTask.java:385)","    ai.h2o.sparkling.extensions.rest.api.ImportFrameHandler.convertColumnsWithTooManyCategoricalLevelsToStringColumns(ImportFrameHandler.scala:84)","    ai.h2o.sparkling.extensions.rest.api.ImportFrameHandler.finalize(ImportFrameHandler.scala:56)","    sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)","    sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)","    sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)","    java.lang.reflect.Method.invoke(Method.java:498)","    water.api.Handler.handle(Handler.java:60)","    water.api.RequestServer.serve(RequestServer.java:470)","    water.api.RequestServer.doGeneric(RequestServer.java:301)","    water.api.RequestServer.doPost(RequestServer.java:227)","    javax.servlet.http.HttpServlet.service(HttpServlet.java:707)","    javax.servlet.http.HttpServlet.service(HttpServlet.java:790)","    ai.h2o.org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)","    ai.h2o.org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:535)","    ai.h2o.org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)","    ai.h2o.org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1317)","    ai.h2o.org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)","    ai.h2o.org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)","    ai.h2o.org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)","    ai.h2o.org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1219)","    ai.h2o.org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)","    ai.h2o.org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)","    ai.h2o.org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)","    water.webserver.jetty9.Jetty9ServerAdapter$LoginHandler.handle(Jetty9ServerAdapter.java:130)","    ai.h2o.org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)","    ai.h2o.org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)","    ai.h2o.org.eclipse.jetty.server.Server.handle(Server.java:531)","    ai.h2o.org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)","    ai.h2o.org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)","    ai.h2o.org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)","    ai.h2o.org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)","    ai.h2o.org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)","    ai.h2o.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)","    ai.h2o.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)","    ai.h2o.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)","    ai.h2o.org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)","    ai.h2o.org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)","    ai.h2o.org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)","    ai.h2o.org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)","    java.lang.Thread.run(Thread.java:748)","Caused by:java.lang.ArrayIndexOutOfBoundsException: 1","    ai.h2o.sparkling.extensions.internals.ConvertCategoricalToStringColumnsTask.map(ConvertCategoricalToStringColumnsTask.java:50)","    water.MRTask.compute2(MRTask.java:675)","    water.MRTask.compute2(MRTask.java:607)","    water.MRTask.compute2(MRTask.java:607)","    water.MRTask.compute2(MRTask.java:607)","    water.MRTask.compute2(MRTask.java:607)","    water.MRTask.compute2(MRTask.java:607)","    water.MRTask.compute2(MRTask.java:607)","    water.H2O$H2OCountedCompleter.compute1(H2O.java:1580)","    ai.h2o.sparkling.extensions.internals.ConvertCategoricalToStringColumnsTask$Icer.compute1(ConvertCategoricalToStringColumnsTask$Icer.java)","    water.H2O$H2OCountedCompleter.compute(H2O.java:1576)","    jsr166y.CountedCompleter.exec(CountedCompleter.java:468)","    jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)","    jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)","    jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)","    jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)"]}
	at ai.h2o.sparkling.backend.utils.RestCommunication$class.checkResponseCode(RestCommunication.scala:326)
	at ai.h2o.sparkling.H2OFrame$.checkResponseCode(H2OFrame.scala:284)
	at ai.h2o.sparkling.backend.utils.RestCommunication$class.readURLContent(RestCommunication.scala:300)
	at ai.h2o.sparkling.H2OFrame$.readURLContent(H2OFrame.scala:284)
	at ai.h2o.sparkling.backend.utils.RestCommunication$class.request(RestCommunication.scala:178)
	at ai.h2o.sparkling.H2OFrame$.request(H2OFrame.scala:284)
	at ai.h2o.sparkling.backend.utils.RestCommunication$class.update(RestCommunication.scala:84)
	at ai.h2o.sparkling.H2OFrame$.update(H2OFrame.scala:284)
	at ai.h2o.sparkling.H2OFrame$.finalizeFrame(H2OFrame.scala:396)
	at ai.h2o.sparkling.backend.Writer$.convert(Writer.scala:103)
	at ai.h2o.sparkling.backend.converters.SparkDataFrameConverter$.toH2OFrame(SparkDataFrameConverter.scala:60)
	at ai.h2o.sparkling.H2OContext$$anonfun$asH2OFrame$2.apply(H2OContext.scala:166)
	at ai.h2o.sparkling.H2OContext$$anonfun$asH2OFrame$2.apply(H2OContext.scala:166)
	at ai.h2o.sparkling.backend.utils.H2OContextExtensions$class.withConversionDebugPrints(H2OContextExtensions.scala:86)
	at ai.h2o.sparkling.H2OContext.withConversionDebugPrints(H2OContext.scala:65)
	at ai.h2o.sparkling.H2OContext.asH2OFrame(H2OContext.scala:166)
	at ai.h2o.sparkling.H2OContext.asH2OFrame(H2OContext.scala:163)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sparklyr.Invoke.invoke(invoke.scala:147)
	at sparklyr.StreamHandler.handleMethodCall(stream.scala:136)
	at sparklyr.StreamHandler.read(stream.scala:61)
	at sparklyr.BackendHandler$$anonfun$channelRead0$1.apply$mcV$sp(handler.scala:58)
	at scala.util.control.Breaks.breakable(Breaks.scala:38)
	at sparklyr.BackendHandler.channelRead0(handler.scala:38)
	at sparklyr.BackendHandler.channelRead0(handler.scala:14)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.chan
```

**Comment by mn-mikke** [comment]
Hi @joscani,
can you share the schema of your dataset?

**Comment by joscani** [comment]
I can't show the name of variables, sorry, but the types using sdf_schema is. 

```
[1] "StringType"        "StringType"        "StringType"        "FloatType"        
  [5] "FloatType"         "FloatType"         "FloatType"         "StringType"       
  [9] "StringType"        "StringType"        "FloatType"         "FloatType"        
 [13] "FloatType"         "StringType"        "FloatType"         "StringType"       
 [17] "StringType"        "IntegerType"       "IntegerType"       "LongType"         
 [21] "IntegerType"       "FloatType"         "LongType"          "LongType"         
 [25] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
 [29] "LongType"          "LongType"          "IntegerType"       "IntegerType"      
 [33] "FloatType"         "FloatType"         "IntegerType"       "IntegerType"      
 [37] "IntegerType"       "FloatType"         "FloatType"         "IntegerType"      
 [41] "IntegerType"       "IntegerType"       "FloatType"         "FloatType"        
 [45] "IntegerType"       "IntegerType"       "IntegerType"       "FloatType"        
 [49] "FloatType"         "IntegerType"       "IntegerType"       "IntegerType"      
 [53] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
 [57] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
 [61] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
 [65] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
 [69] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
 [73] "IntegerType"       "IntegerType"       "IntegerType"       "FloatType"        
 [77] "FloatType"         "IntegerType"       "IntegerType"       "IntegerType"      
 [81] "IntegerType"       "FloatType"         "FloatType"         "IntegerType"      
 [85] "IntegerType"       "FloatType"         "FloatType"         "FloatType"        
 [89] "IntegerType"       "IntegerType"       "IntegerType"       "FloatType"        
 [93] "FloatType"         "FloatType"         "IntegerType"       "IntegerType"      
 [97] "IntegerType"       "FloatType"         "IntegerType"       "IntegerType"      
[101] "IntegerType"       "LongType"          "LongType"          "FloatType"        
[105] "LongType"          "LongType"          "FloatType"         "LongType"         
[109] "IntegerType"       "IntegerType"       "IntegerType"       "IntegerType"      
[113] "FloatType"         "IntegerType"       "FloatType"         "FloatType"        
[117] "LongType"          "DoubleType"        "DoubleType"        "DoubleType"       
[121] "DoubleType"        "LongType"          "LongType"          "LongType"         
[125] "LongType"          "LongType"          "DecimalType(11,1)" "IntegerType"      
[129] "IntegerType"       "StringType"        "StringType"       
``` 
I show a DecimalType(11,1) uhm..

**Comment by mn-mikke** [comment]
There was one bug identified in 3.32.0.1-2 related to the conversion of big datasets. https://h2oai.atlassian.net/browse/SW-2470. Your case/error looks a bit different.

**Comment by joscani** [comment]
Strange. It works with old version.

**Comment by mn-mikke** [comment]
Hi @joscani,
I've managed to reproduce the problem. The fix for it will be a part of the next fix release `3.32.0.2-1`. https://h2oai.atlassian.net/browse/SW-2476

**Comment by joscani** [comment]
wow @mn-mikke  . sparkling water team is amazing. When I have an issue the team fix this quickly.
